File: app/src/main/java/com/mirogal/cocktail/presentation/ui/detail/ingredientlist/IngredientMapper.java
package com.mirogal.cocktail.presentation.ui.detail.ingredientlist;

import com.mirogal.cocktail.presentation.ui.detail.ingredientlist.entity.IngredientEntity;

import java.util.ArrayList;
import java.util.List;

public class IngredientMapper {

    public static List<IngredientEntity> toIngredientList(String[] ingredientList, String[] measureList) {

        List<IngredientEntity> ingredientEntityList = new ArrayList<>();

        int position = 1;
        for (int i = 0; i < ingredientList.length; i++) {
            if (ingredientList[i] != null) {
                ingredientEntityList.add(new IngredientEntity(position, ingredientList[i], measureList[i]));
                position++;
            }
        }

        return ingredientEntityList;
    }
}

File: app/src/main/java/com/mirogal/cocktail/presentation/ui/detail/ingredientlist/IngredientMapper.kt
package com.mirogal.cocktail.presentation.ui.detail.ingredientlist

import com.mirogal.cocktail.presentation.ui.detail.ingredientlist.entity.IngredientEntity
import java.util.*

object IngredientMapper {

    @JvmStatic
    fun toIngredientList(ingredientList: Array<String?>, measureList: Array<String?>): List<IngredientEntity> {
        val ingredientEntityList: MutableList<IngredientEntity> = ArrayList()
        var position = 1
        for (i in ingredientList.indices) {
            if (ingredientList[i] != null) {
                ingredientEntityList.add(IngredientEntity(position, ingredientList[i], measureList[i]))
                position++
            }
        }
        return ingredientEntityList
    }

}